### PR TITLE
Fix query for association

### PR DESCRIPTION
**Pull request by skeleton**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Component/Catalog/Normalizer/Standard/Product/AssociationsNormalizer.php [hunk 1]
```diff
@@ -49,37 +51,56 @@ public function supportsNormalization($data, $format = null)
...
+
+        if (null === $parent) {
+            return [$entityWithFamilyVariant];
+        }
+
+        return array_merge($this->getAncestorProducts($parent), [$entityWithFamilyVariant]);
+    }
+
     /**
      * @param EntityWithAssociationsInterface $associationAwareEntity
```
Comment (anaelChardan) [inline src/Pim/Component/Catalog/Normalizer/Standard/Product/AssociationsNormalizer.php]: The doc is not correct anymore

File: src/Pim/Component/Catalog/Normalizer/Standard/Product/AssociationsNormalizer.php [hunk 2]
```diff
@@ -49,37 +51,56 @@ public function supportsNormalization($data, $format = null)
         return $data instanceof EntityWithAssociationsInterface && 'standard' === $format;
     }
 
+    /**
+     * @param EntityWithFamilyVariantInterface $entityWithFamilyVariant
+     *
+     * @return array
+     */
+    private function getAncestorProducts(EntityWithFamilyVariantInterface $entityWithFamilyVariant): array
```
Comment (anaelChardan) [inline src/Pim/Component/Catalog/Normalizer/Standard/Product/AssociationsNormalizer.php]: :+1:

File: src/Pim/Component/Catalog/Query/AssociatedProduct/GetAssociatedProductCodesByProduct.php [hunk 3]
```diff
@@ -17,5 +17,5 @@
      *
      * @return array
      */
-    public function getCodes(AssociationInterface $association);
+    public function getCodes(int $productId, AssociationInterface $association);
```
Comment (anaelChardan) [inline src/Pim/Component/Catalog/Query/AssociatedProduct/GetAssociatedProductCodesByProduct.php]: update the doc please ;)

Comment (anaelChardan) [reply src/Pim/Component/Catalog/Query/AssociatedProduct/GetAssociatedProductCodesByProduct.php]: The implementations were already changed?

File: src/Pim/Component/Catalog/Normalizer/Standard/Product/AssociationsNormalizer.php [hunk 4]
```diff
@@ -49,37 +51,56 @@ public function supportsNormalization($data, $format = null)
         return $data instanceof EntityWithAssociationsInterface && 'standard' === $format;
     }
 
+    /**
+     * @param EntityWithFamilyVariantInterface $entityWithFamilyVariant
+     *
+     * @return array
```
Comment (anaelChardan) [inline src/Pim/Component/Catalog/Normalizer/Standard/Product/AssociationsNormalizer.php]: @return array|[thetypeoftheitems][]

File: src/Pim/Component/Catalog/Query/AssociatedProduct/GetAssociatedProductCodesByProduct.php (hunk 3 above)
Comment (tamarasaurus) [reply src/Pim/Component/Catalog/Query/AssociatedProduct/GetAssociatedProductCodesByProduct.php]: They were updated here https://github.com/akeneo/pim-enterprise-dev/pull/4492/files

I missed https://github.com/akeneo/pim-community-dev/blob/2.3/src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Query/GetAssociatedProductCodesByProductFromDB.php#L26

**Pull request merged**